documentation/docs/25-build-and-deploy/40-adapter-node.md

```diff
@@ -162,6 +170,58 @@ MY_CUSTOM_ORIGIN=https://my.site \
 node build
 ```
 
+> [Socket activation](#socket-activation) does not work when a prefix is specified because `LISTEN_PID` and `LISTEN_FDS` are set by the OS and cannot be renamed.
+
+## Graceful shutdown
+
+By default `adapter-node` gracefully shuts down the the internal `node:http` server when a `SIGTERM` or `SIGINT` signal is received. It does this by calling [`server.close`](https://nodejs.org/api/http.html#serverclosecallback), [`server.closeIdleConnections`](https://nodejs.org/api/http.html#servercloseidleconnections) and [`server.closeAllConnections`](https://nodejs.org/api/http.html#servercloseallconnections) under the hood. This will
+
+1. wait for requests that have already been made but not received a response yet to finish
+2. reject new requests
+3. close connections once they become idle
+4. and finally, close any remaining connections that are still active after [`SHUTDOWN_TIMEOUT`](#shutdown_timeout) seconds.
+
+> If you want to customize this behaviour you can use a [custom server](#custom-server).
+
+## Socket activation
+
+Most Linux operating systems today use a modern process manager called systemd to start the server and run and manage services. You can configure your server to allocate a socket and start and scale your app on demand. This is called [socket activation](http://0pointer.de/blog/projects/socket-activated-containers.html). In this case the OS will pass two environment variables to your app — `LISTEN_PID` and `LISTEN_FDS`. The adapter will verify that these variables are correct and then listen on file descriptor 3 which refers to a systemd socket unit that you will have to create.
```

Comment:
"will verify that these variables are correct" - is a bit unclear to me. How can they be incorrect? What will happen if they're wrong? Will an error be thrown that the user needs to handle somehow. Or should it be something more like "will check if these variables refer to the current process" because they could be valid, but referring to another process?

Reply:
`LISTEN_PID` should always equal `process.pid`. systemd says you should verify that though I don't know in which cases this condition could ever be false. The second thing that's verified is that only one socket is passed to the app (`LISTEN_FDS === 1`). There isn't really a use case for multiple sockets in a single SvelteKit app. I mentioned it somewhere further down in the docs that only one socket is accepted. If either of those conditions is false then nothing happens. Socket activation wont be enabled and the app just starts as it normally would by listening on `PORT` instead.

Reply:
I feel like it should probably throw an exception rather than silently continuing on in that case

Reply:
What would you suggest? Shoud I just do something like the following?
```js
const socket_activation = listen_pid === process.pid && listen_fds === 1;

if (listen_fds > 1) {
  throw new Error('only one socket is allowed for socket activation')
}
```

Reply:
I was thinking something along the lines of:
```
if (listen_pid !== process.pid) {
  throw new Error(`received LISTEN_PID ${listen_pid} but current process id is ${process.pid}`)
}
if (listen_fds > 1) {
  throw new Error(`only one socket is allowed for socket activation, but LISTEN_FDS was set to ${LISTEN_FDS}`)
}
```

Reply:
Okay, I just added this.